### PR TITLE
ab2d 753 multiple output files per contract - draft - preview

### PR DESCRIPTION
**Pull request by spathiyil**
Enhance Worker  to create multiple output files when the contract has a huge number of patient records instead of creating one massive output file.

### Summary

Enhance Worker  to create multiple output files when the contract has a huge number of patient records instead of creating one massive output file. This is a draft version. Just wanted to give an early preview.

### Checklist

- [ ] Tests and linting pass
- [ ] Code checked for PHI/PII exposure

### Security
N/A

### Additional JIRA Tickets (optional)

N/A

### CODE REVIEW

File: common/src/main/java/gov/cms/ab2d/common/model/JobProgress.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package gov.cms.ab2d.common.model;
```
Comment (dkrylovsb) [inline common/src/main/java/gov/cms/ab2d/common/model/JobProgress.java]: Introducing "job slicing" adds considerable complexity to the domain & processing logic with unclear benefits. We are already processing jobs with the maximum degree of concurrency possible (each patient is an async call to BFD), so slices aren't needed here. Creating slices upfront and them managing them is not really needed to create multiple output files, either; I think there are simpler ways to achieve that.

See how logging frameworks roll log files when they reach a certain threshold. They do so transparently in the background without us knowing it, and of course without having us to "slice" streams of log information. This is exactly what I had in mind. We'd need a `JobOutputWriter` class or similar that encapsulates the file writing/rolling logic.

File: common/src/main/resources/db/changelog/v001/ab2d-753-create-table-job_progress.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+--  -------------------------------------------------------------------------------------------------------------------
+
+--changeset spathiyil:AB2D-753-CreateTable-job_progress failOnError:true
+CREATE TABLE job_progress
+(
+    id                      BIGINT              NOT NULL,
+    job_id                  BIGINT              NOT NULL,
+    contract_id             BIGINT              NOT NULL,
+    slice_number            INTEGER             NOT NULL,
+    records_processed       INTEGER             NOT NULL,
```
Comment (dkrylovsb) [inline common/src/main/resources/db/changelog/v001/ab2d-753-create-table-job_progress.sql]: Slicing makes tracking the overall job progress harder.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/contract/ContractSliceCreatorImpl.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
+package gov.cms.ab2d.worker.processor.contract;
```
Comment (dkrylovsb) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/contract/ContractSliceCreatorImpl.java]: This class would be a good example of the "added complexity" I mentioned above. All the math logic here is pretty hard to follow.

**Comment by dkrylovsb** [comment]
What I would recommend is, creating a new branch on which we could collaborate on designing the `JobOutputWriter` class and see if we could enable multiple output files per job using simpler ways?

**Comment by spathiyil** [comment]
> What I would recommend is, create a new branch on which we'd collaborate on designing the `JobOutputWriter` class and see if we could enable multiple output files per job using simpler ways?

Okay. Declining this branch/pr
